File: ClientRuntimes/Java/src/main/java/com/microsoft/rest/credentials/ServiceClientCredentials.java
/**
 *
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 */

package com.microsoft.rest.credentials;

import com.microsoft.rest.ServiceClient;

/**
 * ServiceClientCredentials is the abstraction for credentials used by ServiceClients accessing REST services.
 */
public abstract class ServiceClientCredentials {

    /**
     * Apply the credentials to the HTTP client builder.
     * @param client The ServiceClient.
     */
    public void applyCredentialsFilter(ServiceClient client) {
        return;
    }
}

File: ClientRuntimes/Ruby/ms-rest/lib/ms_rest/credentials/service_client_credentials.rb
# encoding: utf-8
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

module MsRest
  #
  # Class that serves as a base for all authentications classes.
  #
  class ServiceClientCredentials

    AUTHORIZATION = 'authorization'

    #
    # Base method for performing authentication of HTTP requests.
    # @param request [Net::HTTPRequest] HTTP request to authenticate
    #
    # @return [Net::HTTPRequest] authenticated HTTP request
    def sign_request(request)
      fail ArgumentError, 'request is nil.' if request.nil?
    end
  end
end
